### DISCUSSION TITLE
Regarding low power mode/power saving operation

### DISCUSSION THREAD

**Discussion by ronie19** [earlephilhower/arduino-pico#1135] [General]
I am using RPi Pico with SIM868 GPS/GSM module with bare minimum connections and is powered using a Lithium Ion battery. 
My system sends an SMS, waits for 15 minutes and sends SMS again and waits for 15 mins and so on in a loop. I am driving the system at 20MHz by setting the clock speed in the code. The Pi Pico communicates with SIM868 using UART pins. How low can I go with clock speed ? What other ways are there to conserve power ?
I am using millis method to create delay. What is the best way to conserve power for the Pi Pico in battery driven system where program is in continuos loop.

 ( I am not worried about SIM868 power consumption.)

**Comment by NuclearPhoenixx** [comment]
You could use the [Adafruit SleepyDog Arduino Library](https://github.com/adafruit/Adafruit_SleepyDog), it officially supports the rp2040 architecture. Although, in my tests it didn't really go into sleep (steady power consumption), so it looked more like it just waits much like an standard delay. Maybe I made some mistakes, so YMMV.

**Comment by earlephilhower** [comment]
I think the Adafruit library doesn't actually enable any low power mode.

Right now in the core the is no low power support but you are able to use the native SDK calls.  (My DVM in ammeter mode isn't very accurate at microwatts, so I've not looked into it too deeply.)

Check the pico-extras repo from Raspberry Pi where they have a RTC sleep demo.  You can power down or stop clocks to most units in the SOC, but on wake up it's up to you too make sure they're reenabled and configured properly.  I think you'll need to do some work on the core as well because right now it assumes when it sets something up, like the ADC or PWM, that nothing is going to go and reconfigure it...

**Comment by earlephilhower** [comment]
[Moved to discussions since this isn't related to a core issue and is pretty open ended...]

**Comment by al1fch** [comment]
HI

in the mean time maybe something to do on SIM8xx side if not done yet

I don't use SIM800, not SIM868
After SMS sent  i put SIM800 in sleep mode witn AT+CSCLK=2 (about  2mA ...and SIM800 stay connected to ùobile network)
To awake it i send AT, then immediatly AT+CSCLK=0 to confirm normal mode

**Reply by ronie19** [reply]
So if I wake up the module after 15 mins. Does it immediately locks on GPS and mobile network as well to send a location data through SMS ? How do I confirm that the module has grabbed GPS data and sent the SMS before going back to sleep again ?

**Reply by al1fch** [reply]
for the first point : in AT+CSCLK=2 mode the SIM800 stay connected  to the mobile network and can , for instance,  receive an SMS or vocal call!!

**Reply by ronie19** [reply]
What about GPS fix ?

**Reply by al1fch** [reply]
No GPS with  SIM800  , only GSM and GPRS
and  sorry  for the 'noise'...., I've  just read the end of your initial  message : 
_"I am not worried about SIM868 power consumption"_

(When low power is needed I stay with ESP32 until something comes for that on RP2040 side)
